mobile/src/bundles/common/components/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,57 @@
+// This component was taken from
+// bt-104: Create complete profile screen UI
+// with some minor changes
+
+import CheckBox, { type CheckBoxProps } from '@react-native-community/checkbox';
+import React from 'react';
+import {
+    type Control,
+    type FieldPath,
+    type FieldValues,
+} from 'react-hook-form';
+
+import { Text, View } from '~/bundles/common/components/components';
+import { Color, TextCategory } from '~/bundles/common/enums/enums';
+import { useFormController } from '~/bundles/common/hooks/hooks';
+import { globalStyles } from '~/bundles/common/styles/styles';
+
+type Properties<T extends FieldValues> = CheckBoxProps & {
+    label?: string;
+    name: FieldPath<T>;
+    control: Control<T, null>;
+};
+
+const Checkbox = <T extends FieldValues>({
+    label,
+    name,
+    control,
+    ...props
+}: Properties<T>): JSX.Element => {
+    const { field } = useFormController({ name, control });
+    const { value, onChange } = field;
+
+    const colorCheckbox = props.disabled ? Color.TEXT2 : Color.PRIMARY;
+
```

Comment:
After this [PR](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/119) is approved, we will have an agreed Checkbox component. 
Because I think there is a design mistake and there can't be an additional color for the Checkbox. It can either be enabled or it is disabled and then empty.